### DISCUSSION TITLE
How to set VALVE_MAXIMUM_POSITION?

### DISCUSSION THREAD

**Discussion by sebastianlorenzen** [SukramJ/aiohomematic#1396] [General]
Hello everyone, how can I set the VALVE_MAXIMUM_POSITION from HA? And also how can I read it back?

**Comment by Baxxy13** [comment]
Setzen ist einfach:
![HA_MASTER_Parameter_schreiben](https://github.com/danielperna84/hahomematic/assets/54898249/2de8f0f4-4291-4093-a060-aee7506e23a1)

Lesen auch. Das Problem ist aber das die Methode keinen Einzelparameter zurückgibt sondern den ganzen Parametersatz.
Die Ausgabe müsste dann geparst, oder anderweitig weiterverarbeitet werden. Da muss ich aber passen.
![HA_MASTER_Paramset_lesen](https://github.com/danielperna84/hahomematic/assets/54898249/13a2258a-0a6a-46f2-8962-d1b7bdd23fed)

**Reply by SukramJ** [reply]
@Baxxy13: nicht unignore?

**Reply by Baxxy13** [reply]
Ich denke nicht. 
Das ist kein "State Parameter" sondern liegt im MASTER_PARAMSET. 

Da kommt man nur per XML-RPC oder per JSON-API ran. 
JSON-API wäre von Vorteil da man den Parameter gezielt auslesen könnte, GetParamset holt ja immer den ganzen Parametersatz. 

Edit:
Ich habe aber auch noch nie probiert einen MASTER_PARAMETER zu "unignoren", fällt mir gerade  so auf.

**Reply by SukramJ** [reply]
> Ich habe aber auch noch nie probiert einen MASTER_PARAMETER zu "unignoren", fällt mir gerade so auf.

Ich bin ein wenig überrascht.

`VALVE_MAXIMUM_POSITION@HmIP-eTRV-2:1:MASTER`

**Reply by Baxxy13** [reply]
Ich aber auch...
Das funktioniert ja hervorragend. 😃

**Reply by SukramJ** [reply]
Danke. Die entscheidende Änderung kam aber erst mit der 1.54.0.

**Reply by sebastianlorenzen** [reply]
Cool, danke! Kann ich das irgendwie an das Gerät hängen und alle 24h aktualisieren?

**Reply by sebastianlorenzen** [reply]
Btw, was ist unignore?

**Reply by SukramJ** [reply]
[Diese](https://github.com/danielperna84/custom_homematic#unignore-device-parameters) und andere Fragen erklärt natürlich die [Doku](https://github.com/danielperna84/custom_homematic).

**Reply by sebastianlorenzen** [reply]
Oh wow, garnicht gesehen, dass es so eine umfangreiche Doku gibt. Ich werde mich einlesen.

**Reply by sebastianlorenzen** [reply]
Hallo zusammen,

ich habe versucht überhaupt mal die Parameter Abzufragen. Daran scheitere ich schon.
![image](https://github.com/danielperna84/hahomematic/assets/7313149/0737b050-d266-4865-aa2f-499fce3d2f44)

**Reply by sebastianlorenzen** [reply]
Das Ventil ist ein HM-CC-RT-DN
Die CCU läuft auf einem extra Server nicht mit dem HA zusammen.

**Reply by sebastianlorenzen** [reply]
Muss ich hier noch was eintragen?
In der CCU habe ich die IP-Adresse des HA Servers komplett zugelassen.
![image](https://github.com/danielperna84/hahomematic/assets/7313149/e0d6c3d1-0631-440f-aaf8-4d3cce7275e4)

**Reply by Baxxy13** [reply]
> ich habe versucht überhaupt mal die Parameter Abzufragen. Daran scheitere ich schon.

Falscher Dienst, guck dir meinen Screenshot nochmal genau an.

Außerdem darf beim HM-CC-RT-DN kein Kanal angegeben werden da die Parameter direkt im Gerät liegen (nicht in einem Kanal).

**Reply by sebastianlorenzen** [reply]
Kaum macht man es richtig, geht es :-) Hatte vorher schon Paramset ausgeählt und rumgespielt. Ohne Channel war entscheident.

**Comment by sebastianlorenzen** [comment]
muss ich das so in die unignore datei eintragen?
VALVE_MAXIMUM_POSITION@HM-CC-RT-DN:MASTER 
Oder brauch ich zwei : hintereinander für den Kanal?
Also:
VALVE_MAXIMUM_POSITION@HM-CC-RT-DN::MASTER
oder so?
VALVE_MAXIMUM_POSITION@HM-CC-RT-DN:0:MASTER

**Reply by Baxxy13** [reply]
Aktuell gehen MASTER-Parameter von HM-Geräten nicht mittels unignore.
Wir sind da noch am tüfteln. 
https://github.com/danielperna84/hahomematic/discussions/1397

**Reply by sebastianlorenzen** [reply]
ok, danke. ich habe nur die im Einsatz. 15 davon. Wäre cool, wenn das geht. Ich hab 1397 mal abboniert.
